include/mqtt/async_client.hpp

```diff
@@ -36,7 +36,20 @@ class async_client : public client<Socket, PacketIdBytes> {
      * @param port port number
      * @return async_client object
      */
-    friend std::shared_ptr<callable_overlay<async_client<tcp_endpoint<as::ip::tcp::socket, as::io_context::strand>>>>
+    friend std::shared_ptr<
+        callable_overlay<
+            async_client<
+                tcp_endpoint<
+                    as::ip::tcp::socket,
+#if defined(MQTT_NO_TS_EXECUTORS)
+                    as::strand<as::io_context::executor_type>
+#else
+                    as::io_context::strand
+#endif
```

Comment:
```suggestion
                    MQTT_STRAND
```

And introduce the new header file `strand.hpp` and include it.

```cpp
// Include and header comment like other header files....

#if defined(MQTT_NO_TS_EXECUTORS)
#define MQTT_STRAND  as::strand<as::io_context::executor_type>
#else  // defined(MQTT_NO_TS_EXECUTORS)
#define MQTT_STRAND as::io_context::strand
#endif
```

I want to avoid `#if defined` code repetition.